addons/website_crm_partner_assign/static/src/interactions/crm_partner_assign.js

```diff
@@ -0,0 +1,158 @@
+                state_id: parseInt(this.el.querySelector(".edit_contact_form .state_id").selectedOptions[0].value),
+                country_id: parseInt(this.el.querySelector(".edit_contact_form .country_id").selectedOptions[0].value),
+            },
+        ]);
+        window.location.reload();
+    }
+
+    async createOpportunity() {
+        const response = await this.services.orm.call("crm.lead", "create_opp_portal", [{
+            contact_name: this.el.querySelector(".new_opp_form .contact_name").value,
+            title: this.el.querySelector(".new_opp_form .title").value,
+            description: this.el.querySelector(".new_opp_form .description").value,
+        }]);
+        if (response.errors) {
+            this.el.querySelector("#new-opp-dialog .alert")?.remove();
+            const alertEl = this.el.createElement("div");
+            alertEl.classList.add("alert", "alert-danger");
+            alertEl.textContent = response.errors;
+            const parentEl = this.el.querySelector("#new-opp-dialog");
+            parentEl.insertBefore(alertEl, parentEl.firstElementChild);
+        } else {
+            window.location = "/my/opportunity/" + response.id;
+        }
+    }
+
+    async editOpportunity() {
+        await this.services.orm.call("crm.lead", "update_lead_portal", [
+            [parseInt(this.el.querySelector(".edit_opp_form .opportunity_id").value)],
+            {
+                date_deadline: parse_date(this.el.querySelector(".edit_opp_form .date_deadline").value),
+                expected_revenue: parseFloat(this.el.querySelector(".edit_opp_form .expected_revenue").value),
+                probability: parseFloat(this.el.querySelector(".edit_opp_form .probability").value),
+                activity_type_id: parseInt(this.el.querySelector(".edit_opp_form .next_activity").selectedOptions[0].getAttribute("data")),
+                activity_summary: this.el.querySelector(".edit_opp_form .activity_summary").value,
+                activity_date_deadline: parse_date(this.el.querySelector(".edit_opp_form .activity_date_deadline").value),
+                priority: this.el.querySelector("input[name='PriorityRadioOptions']:checked").value,
+            },
+        ]);
+        window.location.reload();
+    }
+
+    async onInterestedPartnerAssignConfirm() {
+        if (
+            this.el.querySelector(".interested_partner_assign_form .comment_interested").value
+            && this.el.querySelector(".interested_partner_assign_form .contacted_interested").checked
+        ) {
+            await this.confirmInterestedPartner();
+        } else {
+            this.el.querySelector(".interested_partner_assign_form .error_partner_assign_interested").style.display = "block";
+        }
+    }
+
+    async onOppStageButtonClick(ev) {
+        await this.changeOppStage.bind(this, parseInt(ev.currentTarget.getAttribute("opp")), parseInt(ev.currentTarget.getAttribute("data")));
```

Comment:
Fishy... the bound method is not called... also, it is called on `this`, why not a plain call ?